onnxscript/ir/_external_data.py

```diff
@@ -51,3 +70,245 @@ def set_base_dir(graph: _core.Graph | _core.GraphView, base_dir: str | os.PathLi
     for tensor in _all_tensors(graph, include_attributes=True):
         if isinstance(tensor, _core.ExternalTensor):
             tensor.base_dir = base_dir
+
+
+# Loading external data
+
+
+def _load_external_tensor(mmap_buffer, external_tensor):
+    """
+    Load external tensor from mmap buffer
+
+    Args:
+        mmap_buffer: Memory map file buffer.
+        external_tensor: External tensor is to be stored.
+    """
+    # TODO: Make loading function more robust to handle all dtypes
+    raw_data = mmap_buffer.read(external_tensor.nbytes)
+    raw_data_np = np.frombuffer(raw_data, dtype=external_tensor.dtype.numpy())
+    raw_data_np = np.reshape(raw_data_np, newshape=external_tensor.shape.numpy())
+    return raw_data_np
+
+
+def _load_external_data_file(model: _core.Model, file_path: str | os.PathLike) -> None:
+    """
+    Check if file to which external data is to be written to is empty.
+    If file already consists of external data, load external data.
+
+    Args:
+        model: Model to be converted.
+        file_path: Location to which external data is to be stored.
+    """
+    # If file is not empty, load external data
+    with open(file_path, "r+b") as data_file:
+        raw_data = mmap.mmap(
+            data_file.fileno(),
+            0,
+            access=mmap.ACCESS_READ,
+        )
+        for value in model.graph.initializers.values():
+            if isinstance(value.const_value, _core.ExternalTensor):
+                external_tensor = value.const_value
+                external_tensor_path = os.path.join(
+                    external_tensor.base_dir, external_tensor.path
+                )
+                if external_tensor_path != file_path:
+                    continue
+                assert external_tensor.offset is not None
+                data_file.seek(external_tensor.offset)
+
+                raw_data_np = _load_external_tensor(raw_data, external_tensor)
+                tensor = _core.Tensor(
+                    raw_data_np, name=external_tensor.name, dtype=external_tensor.dtype
+                )
+                value.const_value = tensor
+                assert (value.name is not None) and (value.name in model.graph.initializers)
+                model.graph.initializers[value.name] = value
```

Comment:
We don't need to use mmap here because the external tensor class handles loading. To load into memory, simply 

1. Filter out all values with external tensor objects that you want to load into memory
2. Create a new tensor with `external_tensor.numpy()` as data.
3. do the same `value.const_value = new_tensor`
4. `model.graph.initializers[value.name] = value` is not needed as the value was not changed.